src/client/client.go

```diff
@@ -151,6 +151,16 @@ func (c APIClient) DeleteAll() error {
 	return nil
 }
 
+// SetMaxConcurrentStreams Sets the maximum number of concurrent streams the
+// client can have.  This method cannot be safely used be concurrent
+// goroutines.
+func (c APIClient) SetMaxConcurrentStreams(n int) {
+	for i := 0; i < cap(c.streamSemaphore); i++ {
+		c.streamSemaphore <- struct{}{}
+	}
+	c.streamSemaphore = make(chan struct{}, n)
+}
+
```

Comment:
After discussing, its simpler to not have it safe to modify the capacity in flight.